src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -223,6 +224,44 @@ class IndexManagerTests extends SparkFunSuite with SparkInvolvedSuite {
     FileUtils.delete(new Path(refreshTestLocation))
   }
 
+  test("Verify refresh-full rebuild of index in CSV format.") {
+    val refreshTestLocation = sampleParquetDataLocation + "refresh2"
+    FileUtils.delete(new Path(refreshTestLocation))
+    import spark.implicits._
+    SampleData.testData
+      .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      .limit(10)
+      .write
+      .option("header", "true")
+      .csv(refreshTestLocation)
+    val df = spark.read.option("header", "true").csv(refreshTestLocation)
```

Comment:
looks like this is the only difference compared to the parquet test? If so can we refactor common code or do
```scala
Seq(("parquet", None), ("csv", Some(Map("header" -> "true"))).foreach {
  case (format, option) =>
    ...
}
```
Either one is fine as long as duplicate code is removed.

Reply:
While trying to fix this test, I realized that we also need to capture `options` field.
Let me know if there is any other field of HadoopFsRelation that might be useful later..?

```
case class HadoopFsRelation(
    location: FileIndex,
    partitionSchema: StructType,
    dataSchema: StructType,
    bucketSpec: Option[BucketSpec],
    fileFormat: FileFormat,
    options: Map[String, String])(val sparkSession: SparkSession)
```

Reply:
Good catch on capturing `options`! I think that should be all. `partitionSchema` would be populated based on the `rootPaths` during read, so I think we should be good. @apoorvedave1 can double-confirm.